core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,321 @@
+            I64(rhs) => lhs
+                .checked_rem(Decimal::from(rhs))
+                .map(|x| Ok(Decimal(x)))
+                .unwrap_or_else(|| Err(Error::OverflowError("%".to_string()))),
+            F64(rhs) => match Decimal::from_f64_retain(rhs) {
+                Some(x) => lhs
+                    .checked_rem(x)
+                    .map(|y| Ok(Decimal(y)))
+                    .unwrap_or_else(|| Err(Error::OverflowError("%".to_string()))),
+                _ => Err(ValueError::F64ToDecimalConversionError(rhs).into()),
+            },
+            Decimal(rhs) => lhs
+                .checked_rem(rhs)
+                .map(|x| Ok(Decimal(x)))
+                .unwrap_or_else(|| Err(Error::OverflowError("%".to_string()))),
+            Null => Ok(Null),
+            _ => Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(lhs),
+                operator: NumericBinaryOperator::Modulo,
+                rhs: rhs.clone(),
+            }
+            .into()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::{TryBinaryOperator, Value::*},
+        crate::data::{NumericBinaryOperator, ValueError},
+        rust_decimal::prelude::Decimal,
+        std::cmp::Ordering,
+    };
+
+    #[test]
+    fn eq() {
+        let base = Decimal::ONE;
+
+        assert_eq!(base, I8(1));
+        assert_eq!(base, I64(1));
+        assert_eq!(base, F64(1.0));
+        assert_eq!(base, Decimal(Decimal::ONE));
+
+        assert_ne!(base, Bool(true));
+    }
+
+    #[test]
+    fn partial_cmp() {
+        let base = Decimal::ONE;
+
+        assert_eq!(base.partial_cmp(&I8(1)), Some(Ordering::Equal));
+        assert_eq!(base.partial_cmp(&I64(1)), Some(Ordering::Equal));
+        assert_eq!(base.partial_cmp(&F64(1.0)), Some(Ordering::Equal));
+        assert_eq!(
+            base.partial_cmp(&Decimal(Decimal::ONE)),
+            Some(Ordering::Equal)
+        );
+
+        assert_eq!(base.partial_cmp(&Bool(true)), None);
+    }
+
+    #[test]
+    fn try_add() {
+        let base = Decimal::ONE;
+
+        assert!(matches!(base.try_add(&I8(1)), Ok(Decimal(x)) if x == Decimal::TWO ));
+        assert!(matches!(base.try_add(&I64(1)), Ok(Decimal(x)) if x == Decimal::TWO));
+        assert!(matches!(base.try_add(&F64(1.0)), Ok(Decimal(x)) if x == Decimal::TWO));
+        assert!(
+            matches!(base.try_add(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::TWO)
+        );
+
+        assert_eq!(
+            base.try_add(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Add,
+                rhs: Bool(true).clone(),
+            }
+            .into()),
+        );
+    }
+
+    #[test]
+    fn try_subtract() {
+        let base = Decimal::ONE;
+
+        assert!(
+            matches!(base.try_subtract(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ZERO)
+        );
```

Comment:
Even so, the test will pass!

But to give us enough information when debugging, how about using `assert_eq` to know the lhs and rhs values?

Reply:
will do!